src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -439,7 +439,7 @@ private MethodBuilder AddMethodImpl(MethodInfo mi)
                 for (int i = 0; i < parameters.Length; i++)
                 {
                     // args[i] = argi;
-                    if (!parameters[i].IsOut)
+                    if (!parameters[i].IsOut || !parameters[i].ParameterType.IsByRef || parameters[i].IsIn)
```

Comment:
Nit: I'm finding this condition hard to reason about. What would you think about inverting it to something like:
```C#
bool isOutRef = parameters[i].IsOut && parameters[i].ParameterType.IsByRef && !parameters[i].IsIn;
if (!isOutRef)
```
?

Reply:
Agreed, gonna change that.